src/new/wallet.js

```diff
@@ -2781,7 +2782,7 @@ class HathorWallet extends EventEmitter {
       const tokenIdx = tokenUtils.getTokenIndexFromData(token_data);
       const tokenUid = tokens[tokenIdx - 1]?.uid;
       if (!tokenUid) {
-        throw new Error(`Token ${tokenUid} not found in tokens list`);
```

Comment:
if `!tokenUid` means that `tokenUid` is `undefined` or empty string, which makes no sense adding to the error message.
This is why I changed to `token_data` instead, this makes understanding what went wrong easier.